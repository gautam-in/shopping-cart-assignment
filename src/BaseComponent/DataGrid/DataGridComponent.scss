/******************* DataGridComponent *****************************/
@import "../../Styles/color-variable.scss";
@import "../../Styles/variable.scss";

.data-grid {
    width: 100%;
    display: grid;
    gap: 10px;
    margin: 0 10px;
    grid-template-columns: 1fr 1fr 1fr 1fr;

    & .data-grid_item {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-template-rows: 1fr auto 1fr auto;
        gap: 10px;
        width: 100%;
        padding: 10px;
        border-bottom: 2px dotted $LIGHT_GREY;
        box-shadow: 2px 0 2px 0 $LIGHT_GREY;

        & .data-grid_item_header {
            font-size: 14px;
            grid-column: 1/-1;
            align-self: center;
        }

        & .data-grid_item_image {
            width: 100%;
            grid-column: 1/-1;
        }

        & .data-grid_item_description {
            font-size: 10px;
            grid-column: 1/-1;
            background-color: $LIGHT_GREY;
            padding: 5px;
        }

        & .data-grid_item_price {
            font-size: 14px;
            white-space: nowrap;
            letter-spacing: -1px;
            align-self: center;
        }

        & .data-grid_item_button {
            white-space: nowrap;
        }
    }
}