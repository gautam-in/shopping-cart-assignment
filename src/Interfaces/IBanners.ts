export default interface IBanners {
    id: string;
    order: number;
    isActive: boolean;
    bannerImageUrl: string;
    bannerImageAlt: string;
}
