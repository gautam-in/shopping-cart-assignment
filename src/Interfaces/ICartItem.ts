export interface ICartItem {
    id: string;
    name: string;
    imageURL: string;
    price: number;
    unit: number;
}
