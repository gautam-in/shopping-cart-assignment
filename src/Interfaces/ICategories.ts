export interface ICategories {
    id: string;
    order: number;
    enabled: boolean;
    key: string;
    name: string;
    description: string;
    imageUrl: string;
}
