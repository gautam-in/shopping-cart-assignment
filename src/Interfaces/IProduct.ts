export interface IProduct {
    id: string;
    name: string;
    imageURL: string;
    description: string;
    price: number;
    stock: number;
    category: string;
    sku: string;
}
