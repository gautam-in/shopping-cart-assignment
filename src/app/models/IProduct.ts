export interface IProduct{
    "name": string,
    "imageURL": string,
    "description": string,
    "price": number,
    "stock": number,
    "category": string,
    "sku": string,
    "id": string
}