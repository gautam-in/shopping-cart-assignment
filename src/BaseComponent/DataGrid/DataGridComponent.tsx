import { IProduct } from "../../Interfaces/IProduct";
import { addProductToCart } from "../../Model/Model";
import { ROUTE_URL } from "../../Utility/RouteUrlConstant";
import { LinkButtonComponent } from "../LinkButton/LinkButtonComponent";
import "./DataGridComponent.scss";

interface IDataGridComponent {
    productList: Array<IProduct>;
}

export const DataGridComponent: React.FC<IDataGridComponent> = (
    props: IDataGridComponent
) => {
    const BASE_CLASSNAME: string = "data-grid";
    const GRID_ITEM_CLASSNAME: string = `${BASE_CLASSNAME}_item`;
    const GRID_ITEM_HEADER_CLASSNAME: string = `${GRID_ITEM_CLASSNAME}_header`;
    const GRID_ITEM_IMAGE_CLASSNAME: string = `${GRID_ITEM_CLASSNAME}_image`;
    const GRID_ITEM_DESCRIPTION_CLASSNAME: string = `${GRID_ITEM_CLASSNAME}_description`;
    const GRID_ITEM_PRICE_CLASSNAME: string = `${GRID_ITEM_CLASSNAME}_price`;
    const GRID_ITEM_BUTTON_CLASSNAME: string = `${GRID_ITEM_CLASSNAME}_button`;
    return (
        <div className={BASE_CLASSNAME}>
            {props.productList.map((product) => {
                return (
                    <div
                        id={`item_${product.id}`}
                        className={GRID_ITEM_CLASSNAME}
                    >
                        <h3 className={GRID_ITEM_HEADER_CLASSNAME}>
                            {product.name}
                        </h3>
                        <img
                            className={GRID_ITEM_IMAGE_CLASSNAME}
                            src={product.imageURL}
                            alt={product.name}
                        />
                        <p className={GRID_ITEM_DESCRIPTION_CLASSNAME}>
                            {product.description}
                        </p>
                        <p
                            className={GRID_ITEM_PRICE_CLASSNAME}
                        >{`MRP Rs.${product.price}`}</p>
                        <LinkButtonComponent
                            externalClassName={GRID_ITEM_BUTTON_CLASSNAME}
                            id={`btn_${product.id}`}
                            to={ROUTE_URL.CART_PAGE}
                            buttonName={"Buy Now"}
                            onClick={()=>addProductToCart(product.id)}
                        />
                    </div>
                );
            })}
        </div>
    );
};
