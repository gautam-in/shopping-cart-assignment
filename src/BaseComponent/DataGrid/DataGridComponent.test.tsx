import { ShallowWrapper, shallow } from "enzyme";
import { IProduct } from "../../Interfaces/IProduct";
import { DataGridComponent, IDataGridComponent } from "./DataGridComponent";

describe("DataGridComponent", () => {
    const product: IProduct = {
        id: "0",
        name: "0",
        imageURL: "",
        description: "",
        price: 0,
        stock: 0,
        category: "",
        sku: "",
    };

    const productListProps: Array<IProduct> = [product];
    const tree: ShallowWrapper<IDataGridComponent> = shallow(
        <DataGridComponent productList={productListProps} />
    );

    test("DataGridComponent Snapshot render", () => {
        expect(tree).toMatchSnapshot();
    });

    test("Render Child GridComponent based on props", () => {
        expect(tree.children().length).toEqual(productListProps.length)
    });

    test("Check GridComponent Child GridComponent based on props", () => {
        expect(tree.hasClass("data-grid")).toEqual(true)
    });
});
