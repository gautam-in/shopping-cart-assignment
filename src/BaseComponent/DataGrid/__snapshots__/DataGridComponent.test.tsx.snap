// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGridComponent DataGridComponent Snapshot render 1`] = `ShallowWrapper {}`;
